lib/mini_profiler/profiler.rb

```diff
@@ -189,15 +209,23 @@ def call(env)
       # Someone (e.g. Rails engine) could change the SCRIPT_NAME so we save it
       env['RACK_MINI_PROFILER_ORIGINAL_SCRIPT_NAME'] = ENV['PASSENGER_BASE_URI'] || env['SCRIPT_NAME']
 
-      skip_it = (@config.pre_authorize_cb && !@config.pre_authorize_cb.call(env)) ||
-                (@config.skip_paths && @config.skip_paths.any? { |p| path.start_with?(p) }) ||
+      skip_it = (@config.skip_paths && @config.skip_paths.any? { |p| path.start_with?(p) }) ||
                 query_string =~ /pp=skip/
+      if skip_it
+        return client_settings.handle_cookie(@app.call(env))
+      end
+
+      skip_it = (@config.pre_authorize_cb && !@config.pre_authorize_cb.call(env))
 
       if skip_it || (
         @config.authorization_mode == :whitelist &&
         !client_settings.has_valid_cookie?
       )
-        return client_settings.handle_cookie(@app.call(env))
+        if take_snapshot?(path)
+          return client_settings.handle_cookie(take_snapshot(env, start))
```

Comment:
(@SamSaffron This is a very big PR, so I'd like to specifically know if this portion looks OK to you)

From line 212 to this line is the code that determines what requests are eligible for sampling. A request is eligible for sampling if all these conditions are met:

* Request path is NOT in the `skip_paths` config; AND
* `pre_authorize_cb` is not set OR returns `false` OR (authorization mode is whitelist AND the client doesn't have a valid cookie).

This means that requests from users who can see the speed badge will never be sampled. Does this sound OK?

Reply:
interesting, I think this is a fine compromise for v1 of the feature.